[Mobile] - Buttons block - Support content justification

geriux opened the pull request:
## Description
This PR is a continuation of https://github.com/WordPress/gutenberg/pull/23168 adding support of content justification and the `ContentJustificationDropdown` for mobile.

## How has this been tested?
- Open the demo app
- Add a Buttons block
- Add at least two buttons
- Select the block
- **Expect** to see two options in the Toolbar, **Alignment**, and **Content justification**.
- Change the alignment to full-width
- **Expect** to see the block in full-width
- Change the content justification to different options
- **Expect** to see the alignment of the blocks change within the Buttons block

## Screenshots <!-- if applicable -->

ContentJustificationDropdown | ToolbarItem
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/96152469-defbcc00-0f0c-11eb-83c2-a9fb7471021c.gif" width="250" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/96152447-da371800-0f0c-11eb-9100-eb113c6dde0a.png" width="250" /></kbd> 

## Types of changes
New feature

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->

ZebulanStanphill commented on packages/block-library/src/buttons/edit.native.js:
```diff
@@ -31,8 +34,9 @@ function ButtonsEdit( {
+	setAttributes,
 } ) {
-	const { align } = attributes;
+	const { contentJustification } = attributes;
```
You can move this destructuring to the existing destructure for the entire function, e.g.:

```js
function ButtonsEdit( {
	isSelected,
	attributes: { contentJustification },
	onDelete,
	onAddNextButton,
	shouldDelete,
	isInnerButtonSelected,
	setAttributes,
} ) {
```

geriux replied:
Thanks, updated! somehow I had an older version

ZebulanStanphill reviewed (commented):
Test failures appear to be unrelated and caused by test issues that were present in `master` (but now fixed, I think). I've rebased #23168, so rebasing this branch should get tests to pass.

I have no way to actually test the code, but just looking at it, I can't see any mistakes. :+1:

Commit:
Mobile - Buttons block - Support content justification and ContentJustificationDropdown

geriux commented:
> Test failures appear to be unrelated and caused by test issues that were present in `master` (but now fixed, I think). I've rebased #23168, so rebasing this branch should get tests to pass.
> 
> I have no way to actually test the code, but just looking at it, I can't see any mistakes. 👍

Thanks! Just updated the branch
